In [2]:
%matplotlib inline
from IPython.display import Image, HTML
import json
import datetime
import ast
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np
from scipy import stats


In [ ]:
'''
%matplotlib inline
from IPython.display import Image, HTML
import json
import datetime
import ast
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np
from scipy import stats
from sklearn.ensemble import GradientBoostingClassifier, GradientBoostingRegressor
from sklearn.dummy import DummyClassifier, DummyRegressor
from sklearn.model_selection import train_test_split
from xgboost import XGBClassifier, XGBRegressor
from wordcloud import WordCloud, STOPWORDS
import plotly
import plotly.offline as py
py.init_notebook_mode(connected=True)
import plotly.graph_objs as go
import plotly.tools as tls
import warnings
sns.set_style('whitegrid')
sns.set(font_scale=1.25)
pd.set_option('display.max_colwidth', 50)
'''

In [2]:
df8 = pd.read_csv('./datastreams/08_indicadores_macroeconómicos.csv')
df8
## lo que nos muestra que el datastream de indicadores macroeconomicos no es relevante y esta desactualizado
# vistas las otras tablas la informacion llega hasta el año 2022, por eso no se tiene en cuenta.

Año  Trimestre  IPC US  IPC AR USDARS oficial USDARS blue PBI millones
0   2013          1  232773     298           5,00        9,00    2.888.981
1   2013          2  233504     309           5,00        9,00    3.387.811
2   2013          3  234149     331           6,00       10,00    3.436.547
3   2013          4  233049     351           6,00       11,00    3.679.896
4   2014          1  236293     402           8,00       11,00    3.917.649
5   2014          2  238343     433           8,00       13,00    4.702.630
6   2014          3  238031     461           8,00       14,00    4.685.503
7   2014          4  234812     491           9,00       13,00    5.010.564
8   2015          1  236119     518           9,00       13,00    5.092.694
9   2015          2  238638     549           9,00       15,00    5.951.479
10  2015          3  237945     584           9,00       15,00    6.221.731
11  2015          4  236525     621          10,00       15,00    6.552.140
12  2016          1  238132     697          14,00       15,00    6.962.845
13  2016          2  241038     751          14,00       15,00    8.401.125
14  2016          3  241428     791          15,00       16,00    8.448.890
15  2016          4  241432     833          15,00       16,00    8.942.135
16  2017          1  243801     880          16,00       16,00    9.157.377
17  2017          2  244955     925          16,00       18,00   10.595.155
18  2017          3  246819     963          17,00       18,00   10.937.240
19  2017          4  246524    1014          18,00       20,00   11.533.613
20  2018          1  249554    1083          20,00       24,00   11.975.804
21  2018          2  251989    1169          24,00       29,00          NaN

In [3]:
df4 = pd.read_csv('./datastreams/04_accesos_a_internet_fijo_por_tecnología_y_localidad.csv')
df4

Unnamed: 0     Provincia      Partido  \
0              0  BUENOS AIRES   25 de Mayo   
1              1  BUENOS AIRES   25 de Mayo   
2              2  BUENOS AIRES   25 de Mayo   
3              3  BUENOS AIRES   25 de Mayo   
4              4  BUENOS AIRES   25 de Mayo   
...          ...           ...          ...   
3137        3137       TUCUMAN   Tafí Viejo   
3138        3138       TUCUMAN      Trancas   
3139        3139       TUCUMAN      Trancas   
3140        3140       TUCUMAN  Yerba Buena   
3141        3141       TUCUMAN  Yerba Buena   

                               Localidad Link Indec     ADSL CABLEMODEM  \
0                             25 de Mayo    6854100  873.000      4.704   
1                              Del Valle    6854020  181.000          0   
2                      Gobernador Ugarte    6854040    0.000          0   
3                 Norberto de la Riestra    6854060    0.000        782   
4                       Lucas Monteverde    6854050    0.000          0   
...                                  ...        ...      ...        ...   
3137  Villa Mariano Moreno - El Colmenar   90105100    0.000          1   
3138                San Pedro de Colalao   90112020    0.000          0   
3139                   Villa  de Trancas   90112030  150.000          0   
3140                       Villa Carmela   90119020    1.725          0   
3141            Yerba Buena - Marcos Paz   90119030  129.000        203   

      DIAL UP  FIBRA OPTICA  OTROS  SATELITAL  WIMAX  WIRELESS Total general  
0         0.0         2.000    0.0        975      0     664.0         7.218  
1         0.0         0.000   10.0          1      0       0.0           192  
2         0.0         0.000    0.0          0      0     181.0           181  
3         0.0         6.000  167.0          0      0     327.0         1.282  
4         0.0         0.000    0.0          0      0       6.0             6  
...       ...           ...    ...        ...    ...       ...           ...  
3137      0.0       471.000    0.0          0      0       0.0           472  
3138      0.0         0.000    0.0          0      0     313.0           313  
3139      0.0         0.000   33.0          0      0      99.0           282  
3140      0.0         0.000    1.0          0      0      97.0         1.823  
3141      0.0        15.511   33.0          0      0       0.0        15.876  

[3142 rows x 14 columns]

- Se puede observar que en el dataframe habían valores que deben ser cero pero estan registrados - 0 se reemplazan
  con solo el numero 0 pero al final cometi el error de solucionar ese error pero grabe el archivo de exportación con el mismo
  nombre

In [4]:
df4.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3142 entries, 0 to 3141
Data columns (total 14 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Unnamed: 0     3142 non-null   int64  
 1   Provincia      3142 non-null   object 
 2   Partido        3142 non-null   object 
 3   Localidad      3142 non-null   object 
 4   Link Indec     3142 non-null   object 
 5   ADSL           3142 non-null   float64
 6   CABLEMODEM     3142 non-null   object 
 7   DIAL UP        3142 non-null   float64
 8   FIBRA OPTICA   3142 non-null   float64
 9   OTROS          3142 non-null   float64
 10  SATELITAL      3142 non-null   int64  
 11  WIMAX          3142 non-null   int64  
 12  WIRELESS       3142 non-null   float64
 13  Total general  3142 non-null   object 
dtypes: float64(5), int64(3), object(6)
memory usage: 343.8+ KB


In [5]:
df4.columns

Index(['Unnamed: 0', 'Provincia', 'Partido', 'Localidad', 'Link Indec', 'ADSL',
       'CABLEMODEM', 'DIAL UP', 'FIBRA OPTICA', 'OTROS', 'SATELITAL', 'WIMAX',
       'WIRELESS', 'Total general'],
      dtype='object')

- Algunas columnas son object cuando deberían ser float. Es que los puntos están funcionando como separadores de miles. Por lo tanto, los tengo que eliminar para que pasen a valores enteros.

In [6]:
columnas = ['ADSL', 'CABLEMODEM', 'DIAL UP', 'FIBRA OPTICA', 'OTROS', 'SATELITAL','WIMAX', 'WIRELESS' ]

In [7]:
for col in columnas:
    df4[col] = df4[col].replace("nan", '0')    
    df4[col] = df4[col].astype(str).str.replace('.', '')
    df4[col] = pd.to_numeric(df4[col], downcast='integer')

C:\Users\araqu\AppData\Local\Temp\ipykernel_9900\1884540719.py:3: FutureWarning:

The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.



In [8]:
df4.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3142 entries, 0 to 3141
Data columns (total 14 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Unnamed: 0     3142 non-null   int64 
 1   Provincia      3142 non-null   object
 2   Partido        3142 non-null   object
 3   Localidad      3142 non-null   object
 4   Link Indec     3142 non-null   object
 5   ADSL           3142 non-null   int32 
 6   CABLEMODEM     3142 non-null   int32 
 7   DIAL UP        3142 non-null   int16 
 8   FIBRA OPTICA   3142 non-null   int32 
 9   OTROS          3142 non-null   int16 
 10  SATELITAL      3142 non-null   int16 
 11  WIMAX          3142 non-null   int16 
 12  WIRELESS       3142 non-null   int16 
 13  Total general  3142 non-null   object
dtypes: int16(5), int32(3), int64(1), object(5)
memory usage: 214.9+ KB


In [9]:
#Unnamed: 0
df4 = df4.drop('Unnamed: 0', axis=1)

In [12]:
# quedamos con un dataset listo para ser usado por alguna herramienta de visualizacion
df4.to_csv('./accesos_a_internet_fijo_por_tecnología_y_localidad.csv', index=False)

In [13]:
df15 = pd.read_csv('./datastreams/15_total_nacional_de_accesos_a_internet_fijo_por_tipo_de_tecnología.csv')
df15

Año  Trimestre     ADSL  Cablemodem  Fibra óptica   Wireless      Otros  \
0   2022          3  1395277     6031970       2871541     557.11     235.23   
1   2022          2  1468333     5979214       2723285  556243.00  231609.00   
2   2022          1  1533240     6073426       2219533  545814.00  251996.00   
3   2021          4  1657615     5984240       2072236  523107.00  252596.00   
4   2021          3  1950631     5826257       1566048  492415.00     250.19   
5   2021          2  2018587     5641731       1472246  476968.00  253552.00   
6   2021          1  2175211     5424782       1362976  434548.00  250455.00   
7   2020          4  2213949     5371824       1311199  421554.00  253036.00   
8   2020          3  2263889     5259351       1170879  413259.00  248821.00   
9   2020          2  2295533     5005115       1106725  376667.00  247016.00   
10  2020          1  2299457     4903674       1047817  352333.00  209145.00   
11  2019          4  2414575     4883869        941295  340144.00  213298.00   
12  2019          3  2909805     4800091        852221  338241.00  264326.00   
13  2019          2  2990457     4635575        760794  308027.00  265328.00   
14  2019          1  3132789     4435439        637099  276419.00  166389.00   
15  2018          4  3329027     4313831        388013  264299.00  178485.00   
16  2018          3  3521447     4238171        247366  246181.00   93870.00   
17  2018          2  3557216     4038141        226285  194267.00   94535.00   
18  2018          1  3574294     3981129        217460  165300.00   98870.00   
19  2017          4  3584311     3818157        211425  155775.00  100554.00   
20  2017          3  3622575     3670221        193964  106443.00  131740.00   
21  2017          2  3723518     3383434        180777   85813.00   56122.00   
22  2017          1  3708898     3276251        180146   84813.00   55746.00   
23  2016          4  3722794     3210602        178070   85119.00   55088.00   
24  2016          3  3776442     3124855        169898   81455.00   54300.00   
25  2016          2  3782085     3035272        167788   85452.00   59482.00   
26  2016          1  3792493     2806359        164371   85370.00   58763.00   
27  2015          4  3803024     2898226        139187   85726.00   58668.00   
28  2015          3  3788696     2840203        162663   84530.00   58976.00   
29  2015          2  3767821     2756294        150839   82077.00   59157.00   
30  2015          1  3756153     2668248        168188   79098.00   66045.00   
31  2014          4  3764038     2536219        149682   76984.00   71573.00   
32  2014          3  3714764     2569868        155494   85096.00   70049.00   
33  2014          2  3708882     2461670        149363   72405.00   72148.00   
34  2014          1  3697066     2407330        150323   70749.00   72930.00   

       Total        Periodo  
0   11091128  Jul-Sept 2022  
1   10958684   Abr-Jun 2022  
2   10624009   Ene-Mar 2022  
3   10489794   Oct-Dic 2021  
4   10085541  Jul-Sept 2021  
5    9863084   Abr-Jun 2021  
6    9647972   Ene-Mar 2021  
7    9571562   Oct-Dic 2020  
8    9356199  Jul-Sept 2020  
9    9031056   Abr-Jun 2020  
10   8812426   Ene-Mar 2020  
11   8793181   Oct-Dic 2019  
12   9164684  Jul-Sept 2019  
13   8960181   Abr-Jun 2019  
14   8648135   Ene-Mar 2019  
15   8473655   Oct-Dic 2018  
16   8347035  Jul-Sept 2018  
17   8110444   Abr-Jun 2018  
18   8037053   Ene-Mar 2018  
19   7870222   Oct-Dic 2017  
20   7724943  Jul-Sept 2017  
21   7429664   Abr-Jun 2017  
22   7305854   Ene-Mar 2017  
23   7251673   Oct-Dic 2016  
24   7206950  Jul-Sept 2016  
25   7130079   Abr-Jun 2016  
26   6907356   Ene-Mar 2016  
27   6984831   Oct-Dic 2015  
28   6935068  Jul-Sept 2015  
29   6816188   Abr-Jun 2015  
30   6737732   Ene-Mar 2015  
31   6598496   Oct-Dic 2014  
32   6595271  Jul-Sept 2014  
33   6464468   Abr-Jun 2014  
34   6398398   Ene-Mar 2014

In [14]:
df15.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 35 entries, 0 to 34
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Año           35 non-null     int64  
 1   Trimestre     35 non-null     int64  
 2   ADSL          35 non-null     int64  
 3   Cablemodem    35 non-null     int64  
 4   Fibra óptica  35 non-null     int64  
 5   Wireless      35 non-null     float64
 6   Otros         35 non-null     float64
 7   Total         35 non-null     int64  
 8   Periodo       35 non-null     object 
dtypes: float64(2), int64(6), object(1)
memory usage: 2.6+ KB


In [16]:
columnas = ['Otros', 'Wireless' ]
for col in columnas:
    df15[col] = df15[col].replace("nan", '0')    
    df15[col] = df15[col].astype(str).str.replace('.', '')
    df15[col] = pd.to_numeric(df15[col], downcast='integer')

C:\Users\araqu\AppData\Local\Temp\ipykernel_9900\3945872804.py:4: FutureWarning:

The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.



In [17]:
df15.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 35 entries, 0 to 34
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Año           35 non-null     int64 
 1   Trimestre     35 non-null     int64 
 2   ADSL          35 non-null     int64 
 3   Cablemodem    35 non-null     int64 
 4   Fibra óptica  35 non-null     int64 
 5   Wireless      35 non-null     int32 
 6   Otros         35 non-null     int32 
 7   Total         35 non-null     int64 
 8   Periodo       35 non-null     object
dtypes: int32(2), int64(6), object(1)
memory usage: 2.3+ KB


In [2]:
df15 = pd.read_csv('./total_nacional_de_accesos_a_internet_fijo_por_tipo_de_tecnología.csv')

In [5]:
df15 = df15.drop('Periodo', axis=1)

In [6]:
df15.to_csv('./total_nacional_de_accesos_a_internet_fijo_por_tipo_de_tecnología.csv', index=False)

In [8]:
df7 = pd.read_csv('./datastreams/07_conectividad_al_servicio_de_internet.csv')
df7

Unnamed: 0       Partido      Localidad  Unnamed: 3 Unnamed: 4  \
0              NaN  Bahía Blanca  Villa Bordeau         0.0        NaN   
1     BUENOS AIRES           NaN            NaN         0.0         --   
2     BUENOS AIRES           NaN       Balcarce     38376.0         SI   
3     BUENOS AIRES           NaN            NaN       337.0         --   
4     BUENOS AIRES      Balcarce            NaN       374.0         --   
...            ...           ...            ...         ...        ...   
4307         CHACO           NaN      Quitilipi     24517.0         SI   
4308         CHACO     Quitilipi            NaN       879.0         --   
4309         CHACO  San Fernando   Barranqueras         NaN         SI   
4310           NaN  San Fernando         Basail      1929.0        NaN   
4311         CHACO  San Fernando            NaN       336.0         --   

     CABLEMODEM DIALUP Unnamed: 7 Unnamed: 8 WIRELESS TELEFONIAFIJA  \
0            --     --         --        NaN       --            --   
1           NaN    NaN         --         --      NaN           NaN   
2           NaN     SI         SI         --      NaN           NaN   
3           NaN    NaN         --         --      NaN            SI   
4            --    NaN         SI         --       SI            SI   
...         ...    ...        ...        ...      ...           ...   
4307        NaN     --         SI         --      NaN            SI   
4308         --    NaN         --         --       SI            --   
4309         SI     --        NaN         --       SI            SI   
4310         SI     --         --        NaN       SI           NaN   
4311         --    NaN         --         --       --            --   

     Unnamed: 11 Unnamed: 12        link            Latitud        Unnamed: 15  
0            NaN          SI   6056020.0  -38,6472605094596                NaN  
1             --          --         NaN  -38,7765069529222  -62,1851833537179  
2             SI          SI   6063010.0                NaN  -58,2551665841248  
3            NaN          --   6063020.0     -37,9412057603                NaN  
4             SI         NaN   6063030.0  -37,6254980210026  -58,7461862359423  
...          ...         ...         ...                ...                ...  
4307          SI         NaN  22133010.0  -26,8732071732978  -60,2185241226399  
4308         NaN         NaN  22133020.0  -26,4551186226266                NaN  
4309          SI          SI         NaN  -27,4877739289761  -58,9327416886365  
4310          SI          SI         NaN                NaN  -59,2791003619491  
4311          SI         NaN  22140030.0  -27,5620449432874  -59,3096911774268  

[4312 rows x 16 columns]

In [9]:
df7.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4312 entries, 0 to 4311
Data columns (total 16 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Unnamed: 0     2871 non-null   object 
 1   Partido        2866 non-null   object 
 2   Localidad      2884 non-null   object 
 3   Unnamed: 3     2879 non-null   float64
 4   Unnamed: 4     2873 non-null   object 
 5   CABLEMODEM     2862 non-null   object 
 6   DIALUP         2882 non-null   object 
 7   Unnamed: 7     2878 non-null   object 
 8   Unnamed: 8     2872 non-null   object 
 9   WIRELESS       2864 non-null   object 
 10  TELEFONIAFIJA  2873 non-null   object 
 11  Unnamed: 11    2871 non-null   object 
 12  Unnamed: 12    2879 non-null   object 
 13  link           2865 non-null   float64
 14  Latitud        2872 non-null   object 
 15  Unnamed: 15    2879 non-null   object 
dtypes: float64(2), object(14)
memory usage: 539.1+ KB


- si observamos detenidamente la informacion del archivo conectividad_al_servicio_de_internet.csv podemos observar que lo componen 4312 registros pero la mayoría tiene solamente información en 2871 y el que más tiene 2879 lo que significa un 2879/4312 = 0.66 solo un 66% de información por eso no se tendrá en cuenta. 


In [10]:
df10 = pd.read_csv('./datastreams/10_listado_de_localidades_con_conectividad_a_internet.csv')
df10.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4312 entries, 0 to 4311
Data columns (total 12 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Unnamed: 0     2876 non-null   object
 1   Partido        2880 non-null   object
 2   Localidad      2866 non-null   object
 3   ADSL           2868 non-null   object
 4   Unnamed: 4     2876 non-null   object
 5   DIALUP         2888 non-null   object
 6   FIBRAOPTICA    2868 non-null   object
 7   4G             2862 non-null   object
 8   Unnamed: 8     2881 non-null   object
 9   TELEFONIAFIJA  2885 non-null   object
 10  Unnamed: 10    2868 non-null   object
 11  SATELITAL      2878 non-null   object
dtypes: object(12)
memory usage: 404.4+ KB


- Mismo caso que en el dataset anterior listado_de_localidades_con_conectividad_a_internet.csv tampoco se va a utilizar

 - Desde este punto ya se hacen modificaciones a los datasets por los valores observados

In [40]:
df1 = pd.read_csv('./datastreams/01_acceso_a_internet_fijo_por_rangos_de_velocidad_de_bajada_y_provincia.csv')
df1

Año  Trimestre            Provincia  HASTA 512 kbps  \
0    2022          3         Buenos Aires          29.985   
1    2022          3      Capital Federal         517.000   
2    2022          3            Catamarca          71.000   
3    2022          3                Chaco         461.000   
4    2022          3               Chubut         109.000   
..    ...        ...                  ...             ...   
835  2014          1           Santa Cruz         161.000   
836  2014          1             Santa Fe           8.456   
837  2014          1  Santiago Del Estero           1.234   
838  2014          1     Tierra Del Fuego          12.000   
839  2014          1              Tucumán           6.000   

     + 512 Kbps - 1 Mbps + 1 Mbps - 6 Mbps  + 6 Mbps - 10 Mbps  \
0                 27.709           290.315             297.915   
1                  5.742            34.371              67.829   
2                384.000             3.107               5.389   
3                987.000            16.782              18.938   
4                  1.444            45.707              30.940   
..                   ...               ...                 ...   
835                1.625            24.972               1.000   
836              124.468           345.225              20.328   
837               10.531            22.817               2.422   
838              607.000            30.902               6.000   
839               34.672            83.210              11.779   

     + 10 Mbps - 20 Mbps  + 20 Mbps - 30 Mbps  + 30 Mbps    OTROS      Total  
0                267.044              124.190  3.618.689   65.821  4.721.668  
1                 51.946               28.692  1.253.105  105.477  1.547.679  
2                  5.099                3.737     50.298    2.208     70.293  
3                  8.049               15.828     79.390    3.711    144.146  
4                 34.682               15.309     17.563   20.024    165.778  
..                   ...                  ...        ...      ...        ...  
835                1.000                0.000          0    0.000     26.760  
836                6.845               23.000        668    0.000    506.013  
837              109.000                0.000          0    0.000     37.113  
838                0.000                0.000          0    0.000     31.527  
839              362.000                3.000          0    0.000    130.032  

[840 rows x 12 columns]

In [41]:
df1[df1['OTROS'].isna()]

Año  Trimestre         Provincia  HASTA 512 kbps  + 512 Kbps - 1 Mbps  \
313  2019          2   Capital Federal            74.0                1.027   
323  2019          2          La Rioja             9.0               11.000   
324  2019          2           Mendoza            20.0                2.522   
331  2019          2        Santa Cruz            25.0               84.000   
334  2019          2  Tierra Del Fuego             6.0              123.000   
335  2019          2           Tucumán            16.0              157.000   

    + 1 Mbps - 6 Mbps  + 6 Mbps - 10 Mbps  + 10 Mbps - 20 Mbps  \
313           281.370             124.087              239.691   
323            39.819               4.297                4.857   
324           205.165               2.971               10.092   
331            24.861               3.806                1.202   
334            41.824               4.241                1.573   
335            78.891              44.296               24.291   

     + 20 Mbps - 30 Mbps + 30 Mbps  OTROS      Total  
313              296.155   475.991    NaN  1.418.395  
323               13.000     1.298    NaN     50.304  
324                1.256       234    NaN    222.260  
331               28.000         4    NaN     30.010  
334               96.000        32    NaN     47.895  
335              628.000    18.394    NaN    166.673

In [42]:
df1['OTROS'].fillna(0, inplace=True)
df1[df1['OTROS'].isna()]

Empty DataFrame
Columns: [Año, Trimestre, Provincia, HASTA 512 kbps, + 512 Kbps - 1 Mbps, + 1 Mbps - 6 Mbps, + 6 Mbps - 10 Mbps, + 10 Mbps - 20 Mbps, + 20 Mbps - 30 Mbps, + 30 Mbps, OTROS, Total]
Index: []

In [44]:
columnas_int = df1.columns.difference(['Año', 'Trimestre', 'Provincia'])

In [45]:
df1[columnas_int] = df1[columnas_int].applymap( lambda x: str(x).replace('.', ''))

In [46]:
df1[columnas_int] = df1[columnas_int].astype(int)

In [47]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 840 entries, 0 to 839
Data columns (total 12 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   Año                  840 non-null    int64 
 1   Trimestre            840 non-null    int64 
 2   Provincia            840 non-null    object
 3   HASTA 512 kbps       840 non-null    int32 
 4   + 512 Kbps - 1 Mbps  840 non-null    int32 
 5   + 1 Mbps - 6 Mbps    840 non-null    int32 
 6   + 6 Mbps - 10 Mbps   840 non-null    int32 
 7   + 10 Mbps - 20 Mbps  840 non-null    int32 
 8   + 20 Mbps - 30 Mbps  840 non-null    int32 
 9   + 30 Mbps            840 non-null    int32 
 10  OTROS                840 non-null    int32 
 11  Total                840 non-null    int32 
dtypes: int32(9), int64(2), object(1)
memory usage: 49.3+ KB


In [48]:
df1.head()

Año  Trimestre        Provincia  HASTA 512 kbps  + 512 Kbps - 1 Mbps  \
0  2022          3     Buenos Aires           29985                27709   
1  2022          3  Capital Federal            5170                 5742   
2  2022          3        Catamarca             710                 3840   
3  2022          3            Chaco            4610                 9870   
4  2022          3           Chubut            1090                 1444   

   + 1 Mbps - 6 Mbps  + 6 Mbps - 10 Mbps  + 10 Mbps - 20 Mbps  \
0             290315              297915               267044   
1              34371               67829                51946   
2               3107                5389                 5099   
3              16782               18938                 8049   
4              45707                3094                34682   

   + 20 Mbps - 30 Mbps  + 30 Mbps   OTROS    Total  
0                12419    3618689   65821  4721668  
1                28692    1253105  105477  1547679  
2                 3737      50298    2208    70293  
3                15828      79390    3711   144146  
4                15309      17563   20024   165778

In [49]:
df1.to_csv('./acceso_a_internet_fijo_por_rangos_de_velocidad_de_bajada_y_provincia',index=False)

In [15]:
df5 = pd.read_csv('./datastreams/05_accesos_a_internet_fijo_por_velocidad_bajada__y_provincia.csv')
df5

Año  Trimestre            Provincia    Otros 0,256 Mbps 0,375 Mbps  \
0    2022.0        3.0         Buenos Aires   65,821         19        - 0   
1    2022.0        3.0      Capital Federal  105,477         31        - 0   
2    2022.0        3.0            Catamarca    2,208        - 0        - 0   
3    2022.0        3.0                Chaco    3,711        - 0        225   
4    2022.0        3.0               Chubut   20,024          8        - 0   
..      ...        ...                  ...      ...        ...        ...   
476  2017.0        4.0             Santa Fe   14,119        - 0        - 0   
477  2017.0        4.0  Santiago Del Estero      - 0        - 0        - 0   
478  2017.0        4.0     Tierra Del Fuego      - 0        - 0        - 0   
479  2017.0        4.0              Tucumán      - 0        - 0        - 0   
480     NaN        NaN                  NaN      NaN        NaN        NaN   

    0,625 Mbps 0,5 Mbps 0,512 Mbps 0,75 Mbps  ... 70 Mbps 71 Mbps 75 Mbps  \
0          - 0   29,774        192     4,611  ...   3,163     - 0  98,387   
1          - 0      451         35     1,307  ...       3     - 0  30,482   
2          - 0       34         37       - 0  ...     - 0     - 0     - 0   
3          111      234          2       - 0  ...     - 0     - 0     - 0   
4          - 0       69         32       836  ...     - 0     - 0     - 0   
..         ...      ...        ...       ...  ...     ...     ...     ...   
476        - 0    3,109        - 0     7,637  ...     - 0     - 0     - 0   
477        - 0       41        - 0       909  ...     - 0     - 0     - 0   
478        - 0        6        - 0       282  ...     - 0     - 0     - 0   
479        - 0      105        - 0         1  ...     - 0     - 0     - 0   
480        NaN      NaN        NaN       NaN  ...     NaN     NaN     NaN   

    77 Mbps 78 Mbps 80 Mbps 81 Mbps 83 Mbps 82 Mbps 86 Mbps  
0       - 0       2     223     - 0       1     - 0     - 0  
1       - 0     - 0       8     - 0     - 0     - 0     - 0  
2       - 0     - 0     - 0     - 0     - 0     - 0     - 0  
3       - 0     - 0     - 0     - 0     - 0     - 0     - 0  
4       - 0     - 0     - 0     - 0     - 0     - 0     - 0  
..      ...     ...     ...     ...     ...     ...     ...  
476     - 0     - 0     - 0     - 0     - 0     - 0     - 0  
477     - 0     - 0     - 0     - 0     - 0     - 0     - 0  
478     - 0     - 0     - 0     - 0     - 0     - 0     - 0  
479     - 0     - 0     - 0     - 0     - 0     - 0     - 0  
480     NaN     NaN     NaN     NaN     NaN     NaN     NaN  

[481 rows x 97 columns]

In [16]:
df5.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 481 entries, 0 to 480
Data columns (total 97 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Año         480 non-null    float64
 1   Trimestre   480 non-null    float64
 2   Provincia   480 non-null    object 
 3   Otros       480 non-null    object 
 4   0,256 Mbps  480 non-null    object 
 5   0,375 Mbps  480 non-null    object 
 6   0,625 Mbps  480 non-null    object 
 7   0,5 Mbps    480 non-null    object 
 8   0,512 Mbps  480 non-null    object 
 9   0,75 Mbps   480 non-null    object 
 10  1 Mbps      480 non-null    object 
 11  1,2 Mbps    480 non-null    object 
 12  1,25 Mbps   480 non-null    object 
 13  1,5 Mbps    480 non-null    object 
 14  2 Mbps      480 non-null    object 
 15  2,2 Mbps    480 non-null    object 
 16  2,5 Mbps    481 non-null    object 
 17  6,7 Mbps    480 non-null    object 
 18  3 Mbps      480 non-null    object 
 19  3,2 Mbps    480 non-null    o

In [17]:
df5[df5.Año.isna()]
# se comprueba que hay un registros con valores nulos

Año  Trimestre Provincia Otros 0,256 Mbps 0,375 Mbps 0,625 Mbps 0,5 Mbps  \
480  NaN        NaN       NaN   NaN        NaN        NaN        NaN      NaN   

    0,512 Mbps 0,75 Mbps  ... 70 Mbps 71 Mbps 75 Mbps 77 Mbps 78 Mbps 80 Mbps  \
480        NaN       NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   

    81 Mbps 83 Mbps 82 Mbps 86 Mbps  
480     NaN     NaN     NaN     NaN  

[1 rows x 97 columns]

In [18]:
df5 = df5.dropna(subset=['Año', 'Trimestre', 'Provincia'], how= 'all')
# se eliminan todos los valores nulos en las columnas año, trimestre y provincia

In [19]:
df5.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 480 entries, 0 to 479
Data columns (total 97 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Año         480 non-null    float64
 1   Trimestre   480 non-null    float64
 2   Provincia   480 non-null    object 
 3   Otros       480 non-null    object 
 4   0,256 Mbps  480 non-null    object 
 5   0,375 Mbps  480 non-null    object 
 6   0,625 Mbps  480 non-null    object 
 7   0,5 Mbps    480 non-null    object 
 8   0,512 Mbps  480 non-null    object 
 9   0,75 Mbps   480 non-null    object 
 10  1 Mbps      480 non-null    object 
 11  1,2 Mbps    480 non-null    object 
 12  1,25 Mbps   480 non-null    object 
 13  1,5 Mbps    480 non-null    object 
 14  2 Mbps      480 non-null    object 
 15  2,2 Mbps    480 non-null    object 
 16  2,5 Mbps    480 non-null    object 
 17  6,7 Mbps    480 non-null    object 
 18  3 Mbps      480 non-null    object 
 19  3,2 Mbps    480 non-null    o

In [22]:
df5.Año = df5.Año.astype(int)
df5.Trimestre = df5.Trimestre.astype(int)

C:\Users\araqu\AppData\Local\Temp\ipykernel_10660\10919172.py:1: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy

C:\Users\araqu\AppData\Local\Temp\ipykernel_10660\10919172.py:2: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



In [24]:
df5 = df5.applymap(lambda x: 0 if str(x) == '- 0' else x)
# cambio todos los ceros a numero

In [25]:
columnas_flotantes = df5.columns.difference(['Año', 'Trimestre', 'Provincia'])
df5[columnas_flotantes] = df5[columnas_flotantes].applymap(lambda x: str(x).replace(',' , ''))
df5[columnas_flotantes] = df5[columnas_flotantes].applymap(lambda x: str(x).replace('.' , ''))
df5[columnas_flotantes] = df5[columnas_flotantes].astype(int)

In [26]:
df5.head()

Año  Trimestre        Provincia   Otros  0,256 Mbps  0,375 Mbps  \
0  2022          3     Buenos Aires   65821          19           0   
1  2022          3  Capital Federal  105477          31           0   
2  2022          3        Catamarca    2208           0           0   
3  2022          3            Chaco    3711           0         225   
4  2022          3           Chubut   20024           8           0   

   0,625 Mbps  0,5 Mbps  0,512 Mbps  0,75 Mbps  ...  70 Mbps  71 Mbps  \
0           0     29774         192       4611  ...     3163        0   
1           0       451          35       1307  ...        3        0   
2           0        34          37          0  ...        0        0   
3         111       234           2          0  ...        0        0   
4           0        69          32        836  ...        0        0   

   75 Mbps  77 Mbps  78 Mbps  80 Mbps  81 Mbps  83 Mbps  82 Mbps  86 Mbps  
0    98387        0        2      223        0        1        0        0  
1    30482        0        0        8        0        0        0        0  
2        0        0        0        0        0        0        0        0  
3        0        0        0        0        0        0        0        0  
4        0        0        0        0        0        0        0        0  

[5 rows x 97 columns]

In [27]:
df5.to_csv('./accesos_a_internet_fijo_por_velocidad_bajada__y_provincia.csv', index=False)

In [28]:
df6 = pd.read_csv('./datastreams/06_accesos_a_internet_fijo_por_velocidad_de_bajada_y_localidad.csv')
df6

Provincia      Partido                           Localidad  \
0     BUENOS AIRES   25 de Mayo                          25 de Mayo   
1     BUENOS AIRES   25 de Mayo                     Agustín Mosconi   
2     BUENOS AIRES   25 de Mayo                           Del Valle   
3     BUENOS AIRES   25 de Mayo                           Ernestina   
4     BUENOS AIRES   25 de Mayo                   Gobernador Ugarte   
...            ...          ...                                 ...   
3093       TUCUMAN   Tafí Viejo  Villa Mariano Moreno - El Colmenar   
3094       TUCUMAN      Trancas                San Pedro de Colalao   
3095       TUCUMAN      Trancas                   Villa  de Trancas   
3096       TUCUMAN  Yerba Buena                       Villa Carmela   
3097       TUCUMAN  Yerba Buena            Yerba Buena - Marcos Paz   

     Link Indec Otros  0,256 mbps  0,375 mbps 0,5 mbps  0,512 mbps  \
0       6854100   NaN         NaN         NaN        2         NaN   
1       6854010   NaN         NaN         NaN      NaN         NaN   
2       6854020   NaN         NaN         NaN        1         NaN   
3       6854030   NaN         NaN         NaN      NaN         NaN   
4       6854040   NaN         NaN         NaN      NaN         NaN   
...         ...   ...         ...         ...      ...         ...   
3093   90105100   NaN         NaN         NaN      NaN         NaN   
3094   90112020   NaN         NaN         NaN      NaN         NaN   
3095   90112030   NaN         NaN         NaN      NaN         NaN   
3096   90119020   NaN         NaN         NaN      NaN         NaN   
3097   90119030   176         NaN         NaN      NaN         NaN   

      0,625 mbps  ... 75 mbps 78 mbps  80 mbps 82 mbps 83 mbps  85 mbps  \
0            NaN  ...     NaN     NaN      9.0     NaN     NaN      NaN   
1            NaN  ...     NaN     NaN      NaN     NaN     NaN      NaN   
2            NaN  ...     NaN     NaN      NaN     NaN     NaN      NaN   
3            NaN  ...     NaN     NaN      NaN     NaN     NaN      NaN   
4            NaN  ...     NaN     NaN      NaN     NaN     NaN      NaN   
...          ...  ...     ...     ...      ...     ...     ...      ...   
3093         NaN  ...     NaN     NaN      NaN     NaN     NaN      NaN   
3094         NaN  ...     NaN     NaN      NaN     NaN     NaN      NaN   
3095         NaN  ...     NaN     NaN      NaN     NaN     NaN      NaN   
3096         NaN  ...     NaN     NaN      NaN     NaN     NaN      NaN   
3097         NaN  ...     NaN     NaN      NaN     NaN     NaN      NaN   

      90 mbps 92 mbps  95 mbps 100 mbps  
0         NaN     NaN      NaN      NaN  
1         NaN     NaN      NaN      NaN  
2         NaN     NaN      NaN      NaN  
3         NaN     NaN      NaN      NaN  
4         NaN     NaN      NaN      NaN  
...       ...     ...      ...      ...  
3093      NaN     NaN      NaN      304  
3094      NaN     NaN      NaN      NaN  
3095      NaN     NaN      NaN      NaN  
3096      NaN     NaN      NaN      NaN  
3097      NaN     NaN      NaN    6,556  

[3098 rows x 83 columns]

In [29]:
columnas_int = df6.columns.difference(['Provincia', 'Partido', 'Localidad', 'Link Indec'])
columnas_int

Index(['0,256 mbps', '0,375 mbps', '0,5 mbps', '0,512 mbps', '0,625 mbps',
       '0,75 mbps', '1 mbps', '1,25 mbps', '1,5 mbps', '10 mbps', '100 mbps',
       '11 mbps', '12 mbps', '13 mbps', '14 mbps', '15 mbps', '16 mbps',
       '17 mbps', '18 mbps', '19 mbps', '2 mbps', '2,2 mbps', '2,5 mbps',
       '20 mbps', '21 mbps', '22 mbps', '23 mbps', '24 mbps', '25 mbps',
       '25,1 mbps', '25,11 mbps', '25,5 mbps', '26 mbps', '3 mbps', '3,3 mbps',
       '3,5 mbps', '30 mbps', '31 mbps', '32 mbps', '34 mbps', '35 mbps',
       '36 mbps', '38 mbps', '39 mbps', '4 mbps', '4,5 mbps', '40 mbps',
       '41 mbps', '45 mbps', '46 mbps', '49 mbps', '5 mbps', '50 mbps',
       '55 mbps', '58 mbps', '59 mbps', '6 mbps', '6,4 mbps', '60 mbps',
       '61 mbps', '62 mbps', '64 mbps', '65 mbps', '66 mbps', '7 mbps',
       '7,5 mbps', '70 mbps', '75 mbps', '78 mbps', '8 mbps', '80 mbps',
       '82 mbps', '83 mbps', '85 mbps', '9 mbps', '90 mbps', '92 mbps',
       '95 mbps', 'Otros'],
      dtyp

In [34]:
df6[columnas_int] = df6[columnas_int].applymap(lambda x: str(x).replace(',' , ''))
df6[columnas_int] = df6[columnas_int].applymap(lambda x: str(x).replace('.' , ''))

In [35]:
df6

Provincia      Partido                           Localidad  \
0     BUENOS AIRES   25 de Mayo                          25 de Mayo   
1     BUENOS AIRES   25 de Mayo                     Agustín Mosconi   
2     BUENOS AIRES   25 de Mayo                           Del Valle   
3     BUENOS AIRES   25 de Mayo                           Ernestina   
4     BUENOS AIRES   25 de Mayo                   Gobernador Ugarte   
...            ...          ...                                 ...   
3093       TUCUMAN   Tafí Viejo  Villa Mariano Moreno - El Colmenar   
3094       TUCUMAN      Trancas                San Pedro de Colalao   
3095       TUCUMAN      Trancas                   Villa  de Trancas   
3096       TUCUMAN  Yerba Buena                       Villa Carmela   
3097       TUCUMAN  Yerba Buena            Yerba Buena - Marcos Paz   

     Link Indec Otros 0,256 mbps 0,375 mbps 0,5 mbps 0,512 mbps 0,625 mbps  \
0       6854100     0         00         00        2         00         00   
1       6854010     0         00         00        0         00         00   
2       6854020     0         00         00        1         00         00   
3       6854030     0         00         00        0         00         00   
4       6854040     0         00         00        0         00         00   
...         ...   ...        ...        ...      ...        ...        ...   
3093   90105100     0         00         00        0         00         00   
3094   90112020     0         00         00        0         00         00   
3095   90112030     0         00         00        0         00         00   
3096   90119020     0         00         00        0         00         00   
3097   90119030   176         00         00        0         00         00   

      ... 75 mbps 78 mbps 80 mbps 82 mbps 83 mbps 85 mbps 90 mbps 92 mbps  \
0     ...       0      00      90      00      00      00      00      00   
1     ...       0      00      00      00      00      00      00      00   
2     ...       0      00      00      00      00      00      00      00   
3     ...       0      00      00      00      00      00      00      00   
4     ...       0      00      00      00      00      00      00      00   
...   ...     ...     ...     ...     ...     ...     ...     ...     ...   
3093  ...       0      00      00      00      00      00      00      00   
3094  ...       0      00      00      00      00      00      00      00   
3095  ...       0      00      00      00      00      00      00      00   
3096  ...       0      00      00      00      00      00      00      00   
3097  ...       0      00      00      00      00      00      00      00   

     95 mbps 100 mbps  
0         00        0  
1         00        0  
2         00        0  
3         00        0  
4         00        0  
...      ...      ...  
3093      00      304  
3094      00        0  
3095      00        0  
3096      00        0  
3097      00     6556  

[3098 rows x 83 columns]

In [37]:
df6[columnas_int] = df6[columnas_int].fillna(0)
df6[columnas_int] = df6[columnas_int].astype(int)

In [38]:
df6.head()

Provincia     Partido          Localidad Link Indec  Otros  0,256 mbps  \
0  BUENOS AIRES  25 de Mayo         25 de Mayo    6854100      0           0   
1  BUENOS AIRES  25 de Mayo    Agustín Mosconi    6854010      0           0   
2  BUENOS AIRES  25 de Mayo          Del Valle    6854020      0           0   
3  BUENOS AIRES  25 de Mayo          Ernestina    6854030      0           0   
4  BUENOS AIRES  25 de Mayo  Gobernador Ugarte    6854040      0           0   

   0,375 mbps  0,5 mbps  0,512 mbps  0,625 mbps  ...  75 mbps  78 mbps  \
0           0         2           0           0  ...        0        0   
1           0         0           0           0  ...        0        0   
2           0         1           0           0  ...        0        0   
3           0         0           0           0  ...        0        0   
4           0         0           0           0  ...        0        0   

   80 mbps  82 mbps  83 mbps  85 mbps  90 mbps  92 mbps  95 mbps  100 mbps  
0       90        0        0        0        0        0        0         0  
1        0        0        0        0        0        0        0         0  
2        0        0        0        0        0        0        0         0  
3        0        0        0        0        0        0        0         0  
4        0        0        0        0        0        0        0         0  

[5 rows x 83 columns]

In [39]:
df6.to_csv('./accesos_a_internet_fijo_por_velocidad_de_bajada_y_localidad.csv',index=False)

In [ ]:
'''
reemplazos = {
    '2019 *': '2019',
    '2020 *': '2020',
    '2021 *': '2021',
    '2022 *': '2022'
}

tecnologias['Año'] = tecnologias['Año'].replace(reemplazos)
'''

In [ ]:
'''
df3 = pd.read_csv('./datastreams/03_accesos_a_banda_ancha_y_banda_angosta_por_provincia.csv')
df9 = pd.read_csv('./datastreams/09_ingresos_por_la_operación_del_servicio_de_internet_fijo.csv')
df11 = pd.read_csv('./datastreams/11_penetración_del_internet_fijo_por_provincia_(accesos_por_cada_100_habitantes).csv')
df12 = pd.read_csv('./datastreams/12_penetración_del_internet_fijo_por_provincia_(accesos_por_cada_100_hogares).csv')
df14 = pd.read_csv('./datastreams/14_total_nacional_de_accesos_a_internet_fijo_por_banda_ancha_y_banda_angosta.csv')
df16 = pd.read_csv('./datastreams/16_total_nacional_de_accesos_a_internet_fijo_por_velocidad_de_bajada.csv')
df17 = pd.read_csv('./datastreams/16_total_nacional_de_accesos_a_internet_fijo_por_velocidad_de_bajada.csv')
'''